liquibase-standard/src/main/java/liquibase/resource/AbstractResource.java

```diff
@@ -13,16 +16,51 @@ public abstract class AbstractResource implements Resource {
 
     public AbstractResource(String path, URI uri) {
         this.originalPath = path;
-        this.path = path
-                .replace("\\", "/")
-                .replaceFirst("^classpath\\*?:", "")
-                .replaceFirst("^/", "");
-
         if (uri != null) {
-            this.uri = uri.normalize();
+            this.uri = handleUriSyntax(uri);
         } else {
             this.uri = null;
         }
+
+        String pathL = path
+                .replace("\\", "/")
+                .replaceFirst("^classpath\\*?:", "");
```

Comment:
Question: Before doing this replacement here, shouldn't we consider how's set `PRESERVE_CLASSPATH_PREFIX_IN_NORMALIZED_PATHS`?

Reply:
Good question.. this change was already in the code so i considered it to be working before.